web/src/features/rbac/server/membersRouter.ts

```diff
@@ -0,0 +1,818 @@
+                      not: Role.NONE,
+                    },
+                  },
+                },
+              },
+            ],
+          }
+        : {}),
+    },
+    include: {
+      user: {
+        select: {
+          image: true,
+          id: true,
+          name: true,
+          email: true,
+        },
+      },
+    },
+    orderBy: {
+      user: {
+        email: "asc",
+      },
+    },
+    take: query.limit,
+    skip: query.page * query.limit,
+  });
+
+  const totalCount = await prisma.organizationMembership.count({
+    where: {
+      orgId: query.orgId,
+    },
+  });
+
+  const projectMemberships =
+    "projectId" in query
+      ? await prisma.projectMembership.findMany({
+          select: {
+            userId: true,
+            role: true,
+          },
+          where: {
+            orgMembershipId: {
+              in: orgMemberships.map((m) => m.id),
+            },
+            projectId: query.projectId,
+          },
+        })
+      : [];
+
+  return {
+    memberships: orgMemberships.map((om) => ({
+      ...om,
+      projectRole: projectMemberships.find((pm) => pm.userId === om.userId)
+        ?.role,
+    })),
+    totalCount,
+  };
+}
+
+// Extract the function for getting membership invites. It should follow a similar logic than the one for project members. If there is a project ID in the query, I only want to include invites with a specific role for this project. Thank you.
+const orgLevelInviteQuery = z.object({
+  orgId: z.string(),
+  ...paginationZod,
+});
+const projectLevelInviteQuery = orgLevelInviteQuery.extend({
+  projectId: z.string(),
+});
+async function getInvites(
+  prisma: PrismaClient,
+  query:
+    | z.infer<typeof orgLevelInviteQuery>
+    | z.infer<typeof projectLevelInviteQuery>,
+  showAllOrgMembers: boolean = true,
+) {
+  const invitations = await prisma.membershipInvitation.findMany({
+    where: {
+      orgId: query.orgId,
+      // restrict to only invites with role in a project if projectId is set
+      ...("projectId" in query && !showAllOrgMembers
+        ? {
+            OR: [
+              {
+                orgRole: {
+                  not: Role.NONE,
+                },
+              },
+              {
+                projectId: query.projectId,
+                projectRole: {
+                  not: Role.NONE,
+                },
+              },
+            ],
+          }
+        : {}),
+    },
+    include: {
+      invitedByUser: {
+        select: {
+          name: true,
+          image: true,
+        },
+      },
+    },
+    orderBy: {
+      createdAt: "desc",
+    },
+    take: query.limit,
+    skip: query.page * query.limit,
+  });
+
+  const totalCount = await prisma.membershipInvitation.count({
+    where: {
+      orgId: query.orgId,
+    },
+  });
+
+  return {
+    invitations: invitations.map((i) => ({
+      ...i,
+    })),
+    totalCount,
+  };
+}
+
+export const membersRouter = createTRPCRouter({
```

Comment:
router starts in line 246. Can we move stuff from above into different files to make everything more comprehendable?

Reply:
done